design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,174 @@
+
+**Begin Design Discussion**: 2023-11-30
+
+**Status:** draft
+
+## Summary/Abstract
+
+This design proposal outlines the deployment process for the ClusterLink project on a Kubernetes (K8s) cluster.  
+The proposal specifically addresses deployment within the K8s environment.\
+It will rely on a dedicated K8s operator, to make the deployment more user-friendly, simplified and native to K8s users.
+
+## Background
+
+### Motivation and problem space
+
+This design proposal aims to simplify and enhance the deployment of the ClusterLink project.
+By leveraging K8s native mechanisms, such as operators and CRDs, the objective is to establish a user-friendly deployment process.\
+Currently, ClusterLink deployment involves using the cl-adm CLI, which generates a large YAML file that contains the control plane and dataplane deployments, services, and RBAC configurations.\
+In the new design, this process will be managed by the ClusterLink operator, which will consume a simpler and more concise YAML configuration.
+
+### Impact and desired outcome
+
+Implementing this proposal is expected to improve the accessibility and ease of use for ClusterLink users.\
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management.
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+The purpose of ClusterLink is to interconnect various clusters within the same fabric.\
+These clusters may be managed by different owners, located on different cloud providers, and situated in different locations.\
+The deployment of ClusterLink is done by the cluster owner.\
+Each cluster owner is required to obtain the Fabric CA certificate and the cluster certificates (private and public keys) from the fabric administrator.
+
+## Goals
+
+This design document should:
+
+* Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+* Define the task functions of the ClusterLink operator.
+* Define the CRD (Custom Resource Definition) for the ClusterLink operator.
+
+## Non-Goals
+
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+* This document focuses only on the deployment of ClusterLink to the K8s cluster.
+The deployment to another environment as VMs is out of scope.
+* Security considerations related to the creation of certificates for the fabric or each site by the fabric administrator are also beyond the scope of this document
+
+## Proposal
+
+The ClusterLink deployment will be done in the following steps:
+
+1. Deploy the cluster certificates as k8s secrets to the cluster.
+   The certificate include the public Fabric CA certificates and the site certificate that generated by the fabric administrator.
+2. Create the ClusterLink operator.
+3. Deploy the ClusterLink CRD to the operator ClusterLink deployments that includes:
+
+   * ClusterLink controlplane deployment, including controlplane-pod, controlplane-service, and RBAC roles.  
```

Comment:
As worded, this suggests that these fields (e.g., RBAC roles) are part of the CRD.
Did you mean that they're created by the operator in response to the CRD?